recipes-core/disk-encryption/init

```diff
@@ -77,18 +77,14 @@ generate_key() {
 get_key() {
```

Comment:
I suggest to refactor the function to distinguish between successfully reading an empty key vs failing to read it. Otherwise, the error message is misleading.
```
get_key() {
    local key
    key=$(tpm2_nvread $TPM_NV_INDEX 2>>$LOGFILE) || {
        local exit_code=$?
        log "Failed to read TPM NV index (exit code $exit_code). Check $LOGFILE for details."
        return $exit_code
    }
    printf "%s" "$key"
    return 0
}
```